src/main/scala/nl/knaw/dans/bag/v0/FilesXml.scala

```diff
@@ -0,0 +1,165 @@
+package nl.knaw.dans.bag.v0
+
+import java.io.InputStream
+
+import better.files.File
+import nl.knaw.dans.bag.RelativePath
+import nl.knaw.dans.bag.v0.DcTermsElementType.DcTermsElementType
+import nl.knaw.dans.bag.v0.FileAccessCategory.FileAccessCategory
+import nl.knaw.dans.bag.v0.XmlLang.XmlLang
+
+import scala.util.Try
+
+/**
+ * FilesXml gives a listing of all payload files.
+ * the resulting xml file will be valid according to
+ * https://easy.dans.knaw.nl/schemas/bag/metadata/files/files.xsd
+ */
+case class FilesXml() {
+
+  var files : Map[String, FilesXmlItem] = Map()
+  def addFilesXmlItem(item: FilesXmlItem) : Unit = {
+    files += (item.filepath -> item)
+  }
+
+  /**
+   * Saves the deserialized FilesXml at the given location
+   * @param file
+   * @return
+   */
+  def save(file: File): Try[Unit] = ???
+
+}
+
+object FilesXml {
+
+  def empty(): FilesXml = {
+    FilesXml()
+  }
+
+  def read(file: File): Try[FilesXml] = ???
+
+  def read(inputStream: InputStream): Try[FilesXml] = ???
+}
+
+
+/**
+ *
+ * @param pathInData A path in the payload directory of the DansBag
+ * @param fileFormat A format from https://www.iana.org/assignments/media-types/media-types.xhtml
+ * @param accessibleToRights
+ * @param visibleToRights
+ */
+case class FilesXmlItem(pathInData: RelativePath,
+                        fileFormat: FileFormat,
+                        accessibleToRights: Option[FileAccessCategory],
+                        visibleToRights: Option[FileAccessCategory],
+                        dcelements: Set[DcTermsElement] = Set()
+                       ) {
+
+  val filepath: String = pathInData.toString()
+
+}
+case class FileFormat(fileFormat: String)
+
+case class DcTermsElement(tag: DcTermsElementType, value: String, language: Option[XmlLang] = None)
+
+
+
+object FileAccessCategory extends Enumeration {
+  type FileAccessCategory = Value
+
+  val ANONYMOUS: FileAccessCategory = Value
+  val RESTRICTED_GROUP: FileAccessCategory = Value
+  val RESTRICTED_REQUEST: FileAccessCategory = Value
+  val KNOWN: FileAccessCategory = Value
+  val NONE: FileAccessCategory = Value
+}
+
+//subset of the ISO 639-2 language codes
+object XmlLang extends Enumeration {
+  type XmlLang = Value
+
+  val ENG: XmlLang = Value
+  val NLD: XmlLang = Value
+  val FRA: XmlLang = Value
+  val DEU: XmlLang = Value
+}
+
+object DcTermsElementType extends Enumeration {
```

Comment:
This is not limited to FilesXml but also applies to DatasestXml so put it in a separate file. It is not even limited to V0 so put it in another package.